x/concentrated-liquidity/incentives.go

```diff
@@ -193,6 +193,91 @@ func (k Keeper) prepareBalancerPoolAsFullRange(ctx sdk.Context, clPoolId uint64)
 	return canonicalBalancerPoolId, qualifyingFullRangeShares, nil
 }
 
+// nolint: unused
+// claimAndResetFullRangeBalancerPool claims rewards for the "full range" shares corresponding to the given Balancer pool, and
+// then deletes the record from the uptime accumulators. It adds the claimed rewards to the gauge corresponding to the Balancer
+// pool.
+//
+// Returns the number of coins that were claimed and distrbuted.
+// Returns error if either reward claiming, record deletion or adding to the gauge fails.
+func (k Keeper) claimAndResetFullRangeBalancerPool(ctx sdk.Context, clPoolId uint64, balPoolId uint64) (sdk.Coins, error) {
+	// Get CL pool from ID. This also serves as an early pool existence check.
+	clPool, err := k.getPoolById(ctx, clPoolId)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
```

Comment:
I figured since we use `prepareBalancerAsFullRange ` and `claimAndResetFullRangeBalancerPool` in tandem, it would save on state reads/writes (and computation) to have the latter require the caller to specify a valid CL/bal ID pair. The case where the pools are unlinked (or nonexistent) is still caught in this implementation, so from a safety standpoint it's the same (although generally more efficient).